src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -544,9 +544,13 @@ private LocalDefinition EmitInstanceFieldAddress(BoundFieldAccess fieldAccess, A
             //NOTE: we are not propagating AddressKind.Constrained here.
             //      the reason is that while Constrained permits calls, it does not permit 
             //      taking field addresses, so we have to turn Constrained into writeable.
-            var tempOpt = EmitReceiverRef(fieldAccess.ReceiverOpt, addressKind == AddressKind.Constrained ? AddressKind.Writeable : addressKind);
+            var tempOpt = EmitReceiverRef(
+                fieldAccess.ReceiverOpt,
+                field.RefKind == RefKind.None ?
+                    (addressKind == AddressKind.Constrained ? AddressKind.Writeable : addressKind) :
+                    (addressKind != AddressKind.ReadOnlyStrict ? AddressKind.ReadOnly : addressKind));
```

Comment:
We only require a readonly address for the receiver for ref fields since we're loading the field value in those cases. That prevents creating a temporary in cases such as below where ref fields of `in` structs are returned or passed by `ref`. Added a comment.
```C#
ref struct S<T>
{
    public ref T F;
}
class Program
{
    static ref T F1<T>(in S<T> s) { return ref s.F; }
    static void F2<T>(in S<T> s) { M2(ref s.F); }
}
```